src/marker.rs

```diff
@@ -882,6 +866,29 @@ impl<'py> Python<'py> {
     }
 }
 
+impl<'unbound> Python<'unbound> {
```

Comment:
I've taken a liking to properly naming lifetimes as of recently. What would you think of renaming the `Python<'py>` lifetime to `Python<'gil>`? It's still short and I think it represents the meaning of the lifetime better.

Reply:
I agree with the sentiment, but I am not sure if the improvement is worth the churn?

Reply:
It should be a fairly mechanical change if we wanted to update. If we were consistent with always having `'gil` for the band of the GIL lifetime when present, that would also help make it clear when things did not depend on the GIL.